library/std/src/panic.rs

```diff
@@ -285,45 +285,54 @@ where
 
 /// Invokes a closure, capturing the cause of an unwinding panic if one occurs.
 ///
-/// This function will return `Ok` with the closure's result if the closure
-/// does not panic, and will return `Err(cause)` if the closure panics. The
-/// `cause` returned is the object with which panic was originally invoked.
+/// This function will return `Ok` with the closure's result if the closure does
+/// not panic, and will return `Err(cause)` if the closure panics. The `cause`
+/// returned is the object with which panic was originally invoked.
 ///
-/// It is currently undefined behavior to unwind from Rust code into foreign
-/// code, so this function is particularly useful when Rust is called from
-/// another language (normally C). This can run arbitrary Rust code, capturing a
-/// panic and allowing a graceful handling of the error.
+/// It is undefined behavior to unwind from Rust code into foreign code that
+/// does not support unwinding (such as C compiled with `-fno-exceptions`), even
+/// using `extern "C-unwind"`. So this function is particularly useful for Rust
+/// functions that are expected to be called from other languages, because it
+/// can run arbitrary Rust code, capturing a panic and allowing graceful
+/// handling of the error.
```

Comment:
This paragraph is rewritten now, because I somehow missed it on my first pass.

(The changes immediately above and below are simply because the word-wrap was inconsistent; the paragraph above seem to have been at 79, the one below something like 85. I picked 80 and formatted this entire block to be consistent.)

Reply:
The recommended way to expose Rust code to foreign code that does not support unwinding is to use `extern "C"`, right? Then we take care of all the unwind-catching for you. So it seems odd to advise people to use `catch_unwind` for that without also mentioning the more convenient alternative.

Reply:
Please see what you think now.